spec/firebase/firebase_auth_spec.cr

```diff
@@ -1,6 +1,17 @@
 require "../spec_helper"
 
 describe Google::FirebaseAuth do
+  describe "#sign_up" do
+    it "works in case of successful api call" do
+      FirebaseAuthHelper.mock_token
+      FirebaseAuthHelper.mock_sign_up
+
+      expected = Google::FirebaseAuth::SignUpUserResponse.from_json(FirebaseAuthHelper.sign_up_response.to_json).to_json
+      received = FirebaseAuthHelper.firebase.sign_up(email: "test-user@example.com").to_json
+      received.should eq(expected)
```

Comment:
Is the expected type different from the received type?
Any reason they can't be compared directly?
If not, then I would suggest running `JSON.parse` on the `#to_json` output to ensure the comparison doesn't break if the object keys are not ordered in the same way.

Reply:
nope, same type.
Was just copying the style of other tests in this shard.

Reply:
Removing the to_json results in a failure
```
  1) Google::FirebaseAuth #sign_up works in case of successful api call
     Failure/Error: received.should eq(expected)

       Expected: #<Google::FirebaseAuth::SignUpUserResponse:0x7f4b2728bd20 @kind=nil, @id_token=nil, @display_name=nil, @email="test-user@example.com", @refresh_token=nil, @expires_in=nil, @local_id="asdfghjklzxcvbnm">
            got: #<Google::FirebaseAuth::SignUpUserResponse:0x7f4b2728b960 @kind=nil, @id_token=nil, @display_name=nil, @email="test-user@example.com", @refresh_token=nil, @expires_in=nil, @local_id="asdfghjklzxcvbnm">
```

Reply:
ah, you may need to use `def_equals`
OR you can convert it to a `record` which will autogenerate value-level comparison

Reply:
Also records are more suited for immutable data IMO